controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -0,0 +1,385 @@
+		updateMedusaVolumes(dcConfig, medusaSpec, logger)
+	} else {
+		logger.Info("Medusa is not enabled")
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// Generate a secret for Medusa or use the existing one if provided in the spec
+func (r *K8ssandraClusterReconciler) reconcileMedusaSecrets(
+	ctx context.Context,
+	kc *api.K8ssandraCluster,
+	logger logr.Logger,
+) error {
+	logger.Info("Reconciling Medusa user secrets")
+	if kc.Spec.Medusa != nil {
+		cassandraUserSecretRef := kc.Spec.Medusa.CassandraUserSecretRef
+		if cassandraUserSecretRef == "" {
+			cassandraUserSecretRef = cassandraUserSecretName(kc.Spec.Medusa, kc.Name)
+		}
+		logger = logger.WithValues(
+			"MedusaCassandraUserSecretRef",
+			cassandraUserSecretRef,
+		)
+		if err := secret.ReconcileSecret(ctx, r.Client, cassandraUserSecretRef, kc.Name, kc.Namespace); err != nil {
+			logger.Error(err, "Failed to reconcile Medusa CQL user secret")
+			return err
+		}
+	}
+	logger.Info("Medusa user secrets successfully reconciled")
+	return nil
+}
+
+// Create the Medusa config map if it doesn't exist
+func (r *K8ssandraClusterReconciler) reconcileMedusaConfigMap(
+	ctx context.Context,
+	remoteClient client.Client,
+	kc *api.K8ssandraCluster,
+	logger logr.Logger,
+) (ctrl.Result, error) {
+	logger.Info("Reconciling Medusa configMap on namespace : " + kc.Namespace)
+	if kc.Spec.Medusa != nil {
+		medusaIni := createMedusaIni(kc)
+		desiredConfigMap := createMedusaConfigMap(kc.Spec.Medusa, kc.Namespace, kc.Spec.Cassandra.Cluster, medusaIni)
+		// Compute a hash which will allow to compare desired and actual configMaps
+		utils.AddHashAnnotation(desiredConfigMap, api.ResourceHashAnnotation)
+		actualConfigMap := &corev1.ConfigMap{}
+
+		configMapKey := client.ObjectKey{
+			Namespace: kc.Namespace,
+			Name:      fmt.Sprintf("%s-medusa", kc.Spec.Cassandra.Cluster),
+		}
+
+		if err := remoteClient.Get(ctx, configMapKey, actualConfigMap); err != nil {
+			if errors.IsNotFound(err) {
+				if err := remoteClient.Create(ctx, desiredConfigMap); err != nil {
+					logger.Error(err, "Failed to create Medusa ConfigMap")
+					return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
+				}
+			}
+		}
+
+		actualConfigMap = actualConfigMap.DeepCopy()
+
+		if !utils.CompareAnnotations(actualConfigMap, desiredConfigMap, api.ResourceHashAnnotation) {
+			logger.Info("Updating Medusa ConfigMap")
+			resourceVersion := actualConfigMap.GetResourceVersion()
+			desiredConfigMap.DeepCopyInto(actualConfigMap)
+			actualConfigMap.SetResourceVersion(resourceVersion)
+			logger.Info("Updating configMap on namespace " + actualConfigMap.ObjectMeta.Namespace)
+			logger.Info("Should be on namespace " + actualConfigMap.ObjectMeta.Namespace)
+			if err := remoteClient.Update(ctx, actualConfigMap); err != nil {
+				logger.Error(err, "Failed to update Medusa ConfigMap resource")
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
```

Comment:
Maybe we should (and I don't know if that's possible) make it so that Medusa detects changes to this file and reloads the config without requiring another grpc call?